packages/cozy-harvest-lib/src/datacards/GeoDataCard.jsx

```diff
@@ -0,0 +1,229 @@
+
+  L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    attribution:
+      '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+  }).addTo(map)
+
+  return map
+}
+
+const mapStyle = {
+  height: 300
+}
+
+const POMEGRANATE = '#FF0017'
+const EMERALD = '#00D35A'
+const SLATE_GREY = '#313640'
+
+const geojsonMarkerOptions = {
+  radius: 6,
+  color: SLATE_GREY,
+  weight: 1,
+  opacity: 1,
+  fillOpacity: 0.8
+}
+
+const pointToLayer = (feature, latlng) => {
+  return L.circleMarker(latlng, {
+    ...geojsonMarkerOptions,
+    fillColor:
+      feature.properties.feature_type == 'end_place' ? POMEGRANATE : EMERALD
+  })
+}
+
+const geojsonOptions = { pointToLayer }
+
+const TripsMap = ({ trips, index }) => {
+  const nodeRef = useRef()
+  const mapRef = useRef()
+  const tripsRef = useRef([])
+
+  useEffect(() => {
+    mapRef.current = setupMap(nodeRef.current, trips)
+    for (let trip of trips) {
+      const feature = L.geoJSON(trip, geojsonOptions)
+      tripsRef.current.push(feature)
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (mapRef.current) {
+      for (let feature of tripsRef.current) {
+        mapRef.current.removeLayer(feature)
+      }
+      mapRef.current.fitBounds(tripsRef.current[index].getBounds())
+      mapRef.current.addLayer(tripsRef.current[index])
+    }
+  }, [index])
+  return <div style={mapStyle} ref={nodeRef} />
+}
+
+const useCarrousel = (initialState, min, max) => {
+  const [curIndex, setIndex] = useState(0)
+
+  const handlePrev = useCallback(() => {
+    const prev = curIndex - 1
+    setIndex(prev < min ? max : prev)
+  }, [curIndex, min, max])
+
+  const handleNext = useCallback(() => {
+    const next = curIndex + 1
+    setIndex(next > max ? min : next)
+  }, [curIndex, max, min])
+
+  return [curIndex, handlePrev, handleNext]
+}
+
+const TripInfoSlideRaw = ({ trip }) => {
+  return (
+    <div>
+      <Media>
+        <Img>
+          <Icon icon={FlagIcon} color="var(--emerald)" className="u-mr-half" />
+        </Img>
+        <Bd>
+          <Typography variant="body1">
+            {getStartPlaceDisplayName(trip)}
+          </Typography>
+          <Typography variant="caption">
+            {getStartPlaceCaption(trip)}
+          </Typography>
+        </Bd>
+      </Media>
+      <Media>
+        <Img>
+          <Icon
+            icon={FlagIcon}
+            color="var(--pomegranate)"
+            className="u-mr-half"
+          />
+        </Img>
+        <Bd>
+          <Typography variant="body1">
+            {getEndPlaceDisplayName(trip)}
+          </Typography>
+          <Typography variant="caption">{getEndPlaceCaption(trip)}</Typography>
+        </Bd>
+      </Media>
+    </div>
+  )
+}
+
+const TripInfoSlide = memo(TripInfoSlideRaw)
+
+const GeoDataCard = ({ trips }) => {
+  const enhancedTrips = useMemo(() => prepareTrips(trips), [trips])
```

Comment:
The trips are already memoïzed in `DataGeoDataCard`, so is this `useMemo` useful?

Reply:
Solved, I was "preparing" the data twice, now I only do it in the DataGeoCard.